### ISSUE TITLE
WebHook pro změnu stavu platby

### ISSUE THREAD

**Issue by soukicz** [csob/paymentgateway#46]
Je v plánu přidání webhook při změně stavu transakce? Aktuálně je problém, pokud zákazník zavře okno před přesměrováním zpátky do eshopu.

Jediným řešením je pak v intervalech kontrolovat stav všech nedokončených plateb, což je ale značně neoptimální. 

Ideální by bylo posílat URL pro webhook parametrem stejně jako se to dělá s url stránky potvrzení.

**Comment by fprochazka** [comment]
:+1:

**Comment by pepakriz** [comment]
:+1:

**Comment by dmarek** [comment]
ano, webhooky plánujeme, přesně jak píšete, právě pro ošetření zavření prohlížeče před přesměrováním je tento aktivní "push" přístup nejvhodnější.

**Comment by eholy** [comment]
:+1:

**Comment by VasekPurchart** [comment]
:+1:

**Comment by ratajskym** [comment]
Webhook je stále ve feature listu, nicméně v nejbližší době to zařazeno nebude. Task zavírám

**Comment by simPod** [comment]
jakoby core feature si myslim, vote up!

**Comment by soukicz** [comment]
Pravidelně kvůli tomu řešíme duplicitní platby a další problémy. Vypadá to nějak?

**Comment by fprochazka** [comment]
Lol, to furt není?

**Comment by landsman** [comment]
ping

**Comment by Fragous** [comment]
V jakémpak je to asi stavu? Klinická smrt?

**Comment by landsman** [comment]
![tenor 1](https://user-images.githubusercontent.com/3184228/49811798-4ae6d980-fd64-11e8-8c8a-df0246f08528.gif)

**Comment by landsman** [comment]
Hýblo?

**Comment by simPod** [comment]
@landsman *v březnu 2022  
klid

**Comment by Rosnicka** [comment]
Nějaké novoroční předsevzetí? 😉

**Comment by landsman** [comment]
Těch requestů, které musí takto zbytečně odbavovat. Dobrovolný self-DDoS.

![image](https://user-images.githubusercontent.com/3184228/71845946-b3b89680-30c9-11ea-90ba-b62df06eea81.png)
![image](https://user-images.githubusercontent.com/3184228/71845962-bdda9500-30c9-11ea-9cbc-1afdc7e84c9c.png)

**Comment by kukulich** [comment]
Možná se na tom dělá: https://github.com/csob/paymentgateway/issues/456

**Comment by spagr** [comment]
Já bych si "tento rok" vzal na starost pokus o reinkarnaci toho feature requestu. 
Nejen, že absence webhooku u platební brány je v roce 2024 už opravdu ostudná věc, ale
hlavně, jak již bylo zmíněno výše, musí provoz serverů, zvládající dobrovolnou DDoS sebetorturu na endpointu `payment/status`, stát zcela zbytečné náklady ...

Jak to prosím aktuálně vypadá, je možné po těch letech ještě doufat ??

**Comment by matejcekdav** [comment]
Dobrý den, požadavek a potřebu stále (nebo spíše znovu) evidujeme a tudíž doufat ještě můžete. Nicméně v tuto chvíli to časově nejsem schopný nijak ukotvit. Děkuji za pochopení, David M.

**Comment by spagr** [comment]
Děkuji za Vaši reakci. 
Chápu Vaši časovou zaneprázdněnost, nicméně implementace webhooků o provedených platbách na platební bráně ČSOB je pro nás klíčová a prioritní.

Vzhledem k našemu objemu transakcí platební brány v řádu jednotek miliard CZK a stovek tisíc transakcí ročně věříme, že funkce webhooku by měla mít značnou prioritu a může přinést benefity jak pro ČSOB, tak i pro nás jako klienta.

Abychom tuto funkcionalitu posunuli dále, rád bych odkázal naše obchodní oddělení na vhodnou protistranu v ČSOB s cílem získat pro Vás potřebné schválení, prioritu a rozpočet pro její implementaci.

Byli bychom Vám vděční za doporučení, jakým způsobem a na koho se můžeme obrátit, abychom s touto záležitostí co nejefektivněji postoupili.

Předem děkuji

Martin

**Comment by pepakriz** [comment]
@spagr Jen z mé zvědavosti bych se rád zeptal - pokud máte tak velké objemy transakcí, jak se vypořádáváte s dalšími nedodělky čsob brány jako například nemožnost zjistit důvod zamítnutí platby? Nemáte pak zahlcený tech support dotazy zákazníků?

**Comment by matejcekdav** [comment]
@spagr prosím obraťte se na svého obchodního zástupce přiděleného v rámci oddělení Akceptace karet. To je podle mě jediný správný postup v tuto chvíli. Děkuji za pochopení, David

**Comment by CoWayger** [comment]
Je prosím nějaký posun v oblasti webhooku o změně platby? Máme mnoho klientů pod čsob a logy se zbytečně plní poolováním statusů neuhrazených plateb.

**Comment by landsman** [comment]
zadny https://github.com/csob/paymentgateway/issues/693#issuecomment-2426279127

**Comment by landsman** [comment]
> Je prosím nějaký posun v oblasti webhooku o změně platby? Máme mnoho klientů pod čsob a logy se zbytečně plní poolováním statusů neuhrazených plateb.

zadny https://github.com/csob/paymentgateway/issues/693#issuecomment-2426279127
